megamek/src/megamek/common/pathfinder/AeroGroundPathFinder.java

```diff
@@ -0,0 +1,470 @@
+     *  that don't do anything useful.
+     * @param aero The aero entity for which to calculate max thrust.
+     * @return The max thrust.
+     */
+    public static int calculateMaxSafeThrust(IAero aero) {
+        int maxThrust = Math.min(aero.getCurrentThrust(), aero.getSI());    // we should only thrust up to our SI
+        return maxThrust;
+    }
+    
+    /**
+     * 
+     * @author NickAragua
+     *
+     * @param <MovePath> a move path
+     * 
+     * This class removes all off-board paths, but keeps track of the shortest of all the paths removed
+     */
+    public static class AeroGroundOffBoardFilter extends Filter<MovePath> {
+
+        MovePath shortestPath;
+        
+        public MovePath getShortestPath() {
+            return shortestPath;
+        }
+        
+        /**
+         * Returns filtered collection by removing those objects that fail
+         * {@link #shouldStay(T)} test.
+         * 
+         * @param collection collection to be filtered
+         * @return filtered collection
+         */
+        @Override
+        public Collection<MovePath> doFilter(Collection<MovePath> collection) {
+            List<MovePath> filteredMoves = new ArrayList<>();
+            for (MovePath e : collection) {
+                // if the path does not go off board, then we add it to the returned collection
+                if (shouldStay(e)) {
+                    filteredMoves.add(e);
+                }
+                else {
+                    // if the path *does* go off board, we want to compare its length to 
+                    // our current shortest off-board path and keep it in mind if it's both shorter and safe
+                    // at the end of the process, we will add the shortest path off board to the list of moves
+                    if(shortestPath == null || 
+                            (shortestPath.length() > e.length()) && 
+                            AeroPathUtil.isSafePathOffBoard(e)) {
+                        shortestPath = e;
+                    }
+                }
+            }
+            
+            return filteredMoves;
+        }
+        
+        /**
+         * Tests if the object should stay in the collection.
+         * 
+         * @param object tested object
+         * @return true if the object should stay in the collection
+         */
+        @Override
+        public boolean shouldStay(MovePath path) {
+            return !path.fliesOffBoard();
+        }
+        
+    }
+
+    // This object contains a hex and the shortest non-off-board path that has flown over it.
+    HashMap<Coords, MovePath> visitedHexes;
+    
+    /**
+     * Generates paths that begin with all valid acceleration sequences for this aircraft.
+     * Avoids sequences that will cause a PSR (just no)
+     * @param startingPath
+     * @return
+     */
+    public Collection<MovePath> generateValidAccelerations(MovePath startingPath)
+    {
```

Comment:
`public Collection<MovePath> generateValidAccelerations(MovePath startingPath) {`
Also, this can be private.